targets/TARGET_STM/mbed_overrides.c

```diff
@@ -150,5 +150,9 @@ void mbed_sdk_init()
 #endif /* ! MBED_CONF_TARGET_LSE_AVAILABLE */
 #endif /* DEVICE_RTC */
 
+#if defined (MBED_CONF_TARGET_STMOD_SEL) // DISCO_L562QE
```

Comment:
That worries me a bit and it's a bit confusing. `stmod_select` is only defined  for this one DISCO board (not even family or SoC). Here we are conditionally compiling the call to it based on `MBED_CONF_TARGET_STMOD_SEL` and the function is defined as:
```
void stmod_select(void)
{
    DigitalOut SEL_12(PF_11);
    DigitalOut SEL_34(PF_12);

#if (MBED_CONF_TARGET_STMOD_SEL) // UART selection
    SEL_12 = 1;
    SEL_34 = 1;
#else // SPI selection
    SEL_12 = 0;
    SEL_34 = 0;
#endif
}
```
but it's not called anywhere else. I guess it's acceptable, maybe even desirable, for the build to fail if user defines the macro, but the board doesn't provide the function. But maybe it's worth simplifying it a bit? The `stmod_select` will only be called when the macro is defined, maybe the function it doesn't need to have the conditional block? Just a comment saying it's only called when the macro is defined and it will force the UART to be selected?

Reply:
If possible, I'd suggest tweaking the configuration here to pick up the pin names from the [STMOD's mbed_lib.json](
https://github.com/ARMmbed/mbed-os/blob/master/components/cellular/COMPONENT_STMOD_CELLULAR/mbed_lib.json)

Reply:
Agree. Will propose a new solution.